File: internal/web/dto/configmap.go
package dto

type ConfigMapCreateDTO struct {
	BaseInfoDTO
	Data        map[string]string `json:"data"`
	BinaryData  map[string][]byte `json:"binaryData"`
	Annotations map[string]string `json:"annotations"`
	Labels      map[string]string `json:"labels"`
	Yaml        string            `json:"yaml"`
}

type ConfigMapBaseDTO struct {
	ClusterId string `json:"clusterId" form:"clusterId" binding:"required"`
	Namespace string `json:"namespace" form:"namespace"`
	Name      string `json:"name" form:"name"`
	Force     bool   `json:"force" form:"force"`
}

type ConfigMapPageQueryBaseDTO struct {
	PageQueryDTO
	ConfigMapBaseDTO
}

type ConfigMapDataDTO struct {
	BaseInfoDTO
	Data map[string]string `json:"data"`
}

type ConfigMapSettingDTO struct {
	BaseInfoDTO
	Annotations map[string]string `json:"annotations"`
	Labels      map[string]string `json:"labels"`
}

File: front/src/api/configmap.ts
import request, { type ApiResponse, type PageResponse } from '@/utils/request'

export interface ConfigMapPageQueryDTO {
  clusterId: string
  namespace?: string
  page?: number
  pageSize?: number
  keyword?: string
}

export interface ConfigMapBaseDTO {
  clusterId: string
  namespace: string
  name: string
  force?: boolean
}

export interface ConfigMapInfoUpdateDTO {
  clusterId: string
  namespace: string
  name: string
  alias?: string
  describe?: string
}

export interface ConfigMapDataUpdateDTO {
  clusterId: string
  namespace: string
  name: string
  data: Record<string, string>
}

export interface ConfigMapApplyDTO {
  clusterId: string
  yaml: string
}

export interface ConfigMapVO {
  name: string
  namespace: string
  alias?: string
  describe?: string
  data?: Record<string, string>
  annotations?: Record<string, string>
  labels?: Record<string, string>
  creationTimestamp?: string
  yaml?: string
}

export const configMapApi = {
  getList(params: ConfigMapPageQueryDTO): Promise<ApiResponse<PageResponse<ConfigMapVO>>> {
    return request.get('/configMap/list', { params })
  },
  updateInfo(data: ConfigMapInfoUpdateDTO): Promise<ApiResponse<void>> {
    return request.put('/configMap', data)
  },
  updateData(data: ConfigMapDataUpdateDTO): Promise<ApiResponse<void>> {
    return request.put('/configMap/data', data)
  },
  applyYaml(data: ConfigMapApplyDTO): Promise<ApiResponse<void>> {
    return request.post('/configMap/apply', data)
  },
  delete(params: ConfigMapBaseDTO): Promise<ApiResponse<void>> {
    return request.delete('/configMap', { params })
  }
}

export default configMapApi
